be/src/exprs/binary_predicate.cpp

```diff
@@ -468,12 +462,10 @@ DATETIME_BINARY_PRED_FNS()
 #define STRING_BINARY_PRED_FN(CLASS, OP) \
     BooleanVal CLASS::get_boolean_val(ExprContext* ctx, TupleRow* row) { \
         StringVal v1 = _children[0]->get_string_val(ctx, row); \
-        if (v1.is_null) { \
-            return BooleanVal::null(); \
-        } \
         StringVal v2 = _children[1]->get_string_val(ctx, row); \
-        if (v2.is_null) { \
-            return BooleanVal::null(); \
+        BooleanVal result; \
+        if (get_result_for_null(v1, v2, &result)) { \
```

Comment:
I think it's not a good way to add safe is null. I prefer to adding another function to handle this. For performance we should avoid branch and function call here. So you can find the proper function to handle this in Frontend when analyzing.

Reply:
it's inappropriate to add an new operator for BinaryPredicate, because in most conditions we can't know whether the children is null when analyze BinaryPredicate.